gtsfm/evaluation/metrics_report.py

```diff
@@ -42,10 +42,24 @@ def create_table_for_scalar_metrics(metrics_dict: Dict[str, Union[float, int]])
     Returns:
         Table with scalar metrics and their values in HTML format.
     """
-    table = {"Metric name": list(metrics_dict.keys()), "Value": list(metrics_dict.values())}
+    if "gtsfm" in list(metrics_dict.keys())[0] or "colmap" in list(metrics_dict.keys())[0]:
```

Comment:
this is not a clean way to write this code. Ideally, we'd like to have a function that gets colmap metrics and gtsfm metrics separately, why are the two combined into the same metrics dict when they are from different sources?

Reply:
In generate_metrics_report_html(), the tables and plots are created in a for-loop over each metric group, with get_figures_for_metrics(metrics_group), which doesn't support storing those metrics separately.

Would it be cleaner to write an additional function for getting_figures_for_multiple_metrics([metrics_group, ...]), then add the heading and table outside of the loop where the rest of the metric groups' headings and tables are written?